sdk/eventhub/event-hubs/src/eventData.ts

```diff
@@ -214,8 +215,8 @@ export namespace EventDataInternal {
     if (data.applicationProperties) {
       msg.application_properties = data.applicationProperties;
     }
-    if (data.partitionKey != undefined) {
-      msg.message_annotations[Constants.partitionKey] = data.partitionKey;
+    if (partitionKey != undefined) {
+      msg.message_annotations[Constants.partitionKey] = partitionKey;
```

Comment:
I wonder if it is enough to set the partitionKey on the outer batch message rather than each event inside the batch. Can we test it out? If that is true, then we don't have to pass the partitionKey into this function at all.

Reply:
Yes, it is enough to set the partitionKey on the outer batch message. we don't need to set partitionKey for each event.

If we set the partitionKey on the outer batch message, all events go to only one partition.

Reply:
As per offline discussions, we don't have a reliable way of testing this because the promise of round robin in the absence of partitionKey is not reliable. We will go with what we have now.